zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/incident/IncidentUpdateRepositoryIT.java

```diff
@@ -53,59 +52,142 @@
 
 @Testcontainers
 @AutoCloseResources
-final class ElasticsearchIncidentUpdateRepositoryIT {
-
-  @Container
-  private static final ElasticsearchContainer CONTAINER =
-      TestSearchContainers.createDefeaultElasticsearchContainer();
-
+abstract sealed class IncidentUpdateRepositoryIT {
   private static final int PARTITION_ID = 1;
   private static final Duration REQUEST_TIMEOUT = Duration.ofSeconds(5);
 
-  @AutoCloseResource private final RestClientTransport transport = createRestClient();
   private final String indexPrefix = UUID.randomUUID().toString();
-  private final ElasticsearchClient testClient = new ElasticsearchClient(transport);
-  private final ElasticsearchEngineClient searchClient = new ElasticsearchEngineClient(testClient);
-  private final ElasticsearchAsyncClient client = new ElasticsearchAsyncClient(transport);
-  private final PostImporterQueueTemplate postImporterQueueTemplate =
+  protected final PostImporterQueueTemplate postImporterQueueTemplate =
       new PostImporterQueueTemplate(indexPrefix, true);
-  private final IncidentTemplate incidentTemplate = new IncidentTemplate(indexPrefix, true);
-  private final ListViewTemplate listViewTemplate = new ListViewTemplate(indexPrefix, true);
-  private final FlowNodeInstanceTemplate flowNodeInstanceTemplate =
+  protected final IncidentTemplate incidentTemplate = new IncidentTemplate(indexPrefix, true);
+  protected final ListViewTemplate listViewTemplate = new ListViewTemplate(indexPrefix, true);
+  protected final FlowNodeInstanceTemplate flowNodeInstanceTemplate =
       new FlowNodeInstanceTemplate(indexPrefix, true);
 
+  @AutoCloseResource private final ClientAdapter clientAdapter;
+  private final SearchEngineClient engineClient;
+
+  protected IncidentUpdateRepositoryIT(final String databaseUrl) {
+    final var config = new ExporterConfiguration();
+    config.getConnect().setIndexPrefix(indexPrefix);
+    config.getConnect().setUrl(databaseUrl);
+
+    clientAdapter = ClientAdapter.of(config);
+    engineClient = clientAdapter.getSearchEngineClient();
+  }
+
   @BeforeEach
   void beforeEach() {
     Stream.of(
             postImporterQueueTemplate, incidentTemplate, listViewTemplate, flowNodeInstanceTemplate)
-        .forEach(template -> searchClient.createIndexTemplate(template, new IndexSettings(), true));
+        .forEach(template -> engineClient.createIndexTemplate(template, new IndexSettings(), true));
   }
 
-  private ElasticsearchIncidentUpdateRepository createRepository() {
-    return new ElasticsearchIncidentUpdateRepository(
-        PARTITION_ID,
-        postImporterQueueTemplate.getAlias(),
-        incidentTemplate.getAlias(),
-        listViewTemplate.getAlias(),
-        flowNodeInstanceTemplate.getAlias(),
-        client,
-        Runnable::run);
+  protected abstract IncidentUpdateRepository createRepository();
+
+  static final class ElasticsearchIT extends IncidentUpdateRepositoryIT {
```

Comment:
:question: Do we need to add `@AutoCloseResource` here as well since it's a static inner class?

Reply:
The annotation is inherited, so no need.